app/jobs/write_chapter_content_job.rb

```
@@ -0,0 +1,117 @@
+  def perform(chapter_id, previous_chapter_id)
+    @component = Chapter.find(chapter_id)
+    @previous_chapter = Chapter.find(previous_chapter_id) if previous_chapter_id
+    @book = @component.book
+    @pages_per_chapter = calculate_pages_per_chapter
+    @target_words = @pages_per_chapter * WORDS_PER_PAGE
+    
+    super()
+    
+    content_response = send_chat_request
+    chapter_content = JSON.parse(content_response['choices'][0]['message']['content'])['content']
+    
```

RuboCop (1.69.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)